share/spack/setup-env.sh

```diff
@@ -121,20 +121,28 @@ function spack {
             # If spack module command comes back with an error, do nothing.
             case $_sp_subcommand in
                 "use")
-                    if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type dotkit "${_sp_spec[@]}"); then
+                    if _sp_full_spec=$(command spack $_sp_flags module dotkit find $_sp_subcommand_args "${_sp_spec[@]}"); then
                         use $_sp_module_args $_sp_full_spec
+                    else
+                        $(exit 1)
                     fi ;;
                 "unuse")
-                    if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type dotkit "${_sp_spec[@]}"); then
+                    if _sp_full_spec=$(command spack $_sp_flags module dotkit find $_sp_subcommand_args "${_sp_spec[@]}"); then
                         unuse $_sp_module_args $_sp_full_spec
+                    else
+                        $(exit 1)
                     fi ;;
                 "load")
-                    if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type tcl "${_sp_spec[@]}"); then
+                    if _sp_full_spec=$(command spack $_sp_flags module tcl find $_sp_subcommand_args "${_sp_spec[@]}"); then
```

Comment:
This keeps the existing behavior, that is, `spack load spec` only loads `spec` and `spack load -r spec` loads `spec` and all of its dependencies.

I agree that autoloading (at least) the run dependencies by default makes sense. I have been using `autoload: 'all'` for quite some time for our installation since users kept forgetting to use `-r`. Maybe a new `autoload` option would make sense for this?

Reply:
The issue where we discussed this was #8639.

Reply:
> Is this PR a decent place to fix that or is this something you want to save for later?

If it is fine with yous I'd save this for a later (and smaller) PR. In any case I plan to continue working on module issues in the next couple of weeks.

Reply:
> doesn't this and the rest of d4b4be4 get rid of the loading of dependencies?

No, this keeps the current behavior, except that it prevents `spack load` from loading multiple versions of the same package (see #6676).